opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQASTNode.hpp

```diff
@@ -62,12 +62,16 @@ class UDQASTNode {
 
     bool operator==(const UDQASTNode& data) const;
 
-    const std::string& stringValue() const;
-    UDQTokenType getType() const;
-    double scalarValue() const;
-    const std::vector<std::string>& getSelectors() const;
-    const std::shared_ptr<UDQASTNode>& getLeft() const;
-    const std::shared_ptr<UDQASTNode>& getRight() const;
+    template<class Serializer>
+    void serializeOp(Serializer& serializer)
+    {
+        serializer(type);
+        serializer(string_value);
+        serializer(selector);
+        serializer(scalar_value);
+        serializer(left);
+        serializer(right);
```

Comment:
These shared pointers, make me a bit uneasy. If there is a reason (storage at multiple places, pointing to one object) for them, then this will be lost (each place will have its own different object), if I am not mistaking.

Reply:
They are only shared_ptr for have-a-value or not reasons, could have been unique as such.

Reply:
it's actually quite messy. they are created using make_unique, but kept as shared pointers because instances of UDQASTNode is passed as value here and there (return values..). but at the end of the day there is just a single copy of the pointers, i have carefully checked by printing the ref counts etc.

Reply:
Could have been just stored by value, then.

Reply:
indeed. but it's not. could be an improvement for later.